src/antsibull_nox/sessions/execution_environment.py

```diff
@@ -0,0 +1,275 @@
+        "--file",
+        "execution-environment.yml",
+        "--tag",
+        image_name,
+        "--container-runtime",
+        container_engine,
+        "--verbosity",
+        "3",
+        "--context",
+        str(directory),
+    ]
+    with session.chdir(directory):
+        session.run(*cmd)  # , silent=True)
+    return image_name
+
+
+def prepare_execution_environment(
+    *,
+    session: nox.Session,
+    execution_environment: ExecutionEnvironmentData,
+    container_engine: str,
+) -> tuple[Path | None, str | None, CollectionData]:
+    """
+    Generate execution environments for a collection.
+
+    Args:
+        session: Nox session object
+        execution_environment: EE configuration data
+        container_engine: Container runtime to use
+
+    Returns:
+        Tuple with:
+        - collection_tarball_path: Path to the built collection tarball
+        - built_image_names: List of built container images
+        - collection_data: Collection metadata
+    """
+    collection_tarball_path, collection_data, _ = build_collection(session)
+
+    if collection_tarball_path is None:
+        return collection_tarball_path, None, collection_data
+
+    directory = Path(session.create_tmp()) / "ee"
+    if directory.is_dir():
+        shutil.rmtree(directory)
+    if not directory.is_dir():
+        directory.mkdir()
+
+    generate_ee_config(
+        directory=directory,
+        collection_tarball_path=collection_tarball_path.absolute(),
+        collection_data=collection_data,
+        ee_config=execution_environment.config,
+    )
+
+    built_image = build_ee_image(
+        session=session,
+        directory=directory,
+        ee_name=execution_environment.name,
+        collection_data=collection_data,
+        container_engine=container_engine,
+    )
+
+    return collection_tarball_path, built_image, collection_data
+
+
+def add_execution_environment_session(
+    *,
+    session_name: str,
+    execution_environment: ExecutionEnvironmentData,
+    container_engine: str,
+    default: bool = False,
+) -> None:
+    """
+    Build and test execution environments for the collection.
+    """
+
+    def session_func(session: nox.Session):
+        install(session, "ansible-builder", "ansible-navigator")
+
+        collection_tarball, built_image, collection_data = (
+            prepare_execution_environment(
+                session=session,
+                execution_environment=execution_environment,
+                container_engine=container_engine,
+            )
+        )
+
+        if collection_tarball is None or built_image is None:
+            # Install only
+            return
+
+        session.log(
+            f"Building execution environment {execution_environment.description}"
+            f" for {collection_data.namespace}.{collection_data.name}. Image: {built_image}"
+            f" using {container_engine}"
+        )
+
+        session.run("/bin/sh", "-c", "pwd && ls -lah")
```

Comment:
This looks like debug output to me; if it is, it should be removed before merging:
```suggestion
```